File: Medium/aneo/aneo.py
import sys
import math

speed = int(input())
light_count = int(input())
lights = [[*map(int,input().split())]for i in range(light_count)]

while True:
    currentSpeed = speed
    for distance, duration in lights:
        timeToArrive = 3.6 * distance/speed
        if (timeToArrive / duration) % 2 >= 1:
            speed-=1
            break
    if currentSpeed == speed:
        break
print(speed)

File: Medium/aneo/aneo.js
let speed = parseInt(readline());
const lightCount = parseInt(readline());
const lights = [];
for (let i = 0; i < lightCount; i++) {
    lights.push(readline().split(' ').map(Number));
}


while (true) {
    const currentSpeed = speed;
    for([distance, duration] of lights) {
        const timeToArrive = 3.6 * distance / speed;
        if ((timeToArrive / duration) % 2 >= 1){
            speed--;
            break;
        }

    }
    if (currentSpeed === speed) break;
}

console.log(speed)